docs/operators/ftpInputOperator.md

```diff
@@ -0,0 +1,54 @@
+FTP Input Operator
+=============
+
+## Operator Objective
+This operator is designed to scan a directory from an FTP server for files, read
+and split file content into tuples such as lines or blocks of bytes, and finally
+emit them on the output port for further processing by downstream operators.
+The operator extends the AbstractFileInputOperator. It overrides the
+getFSInstance() method and returns an instance of the FTPFileSystem
+(org.apache.hadoop.fs.ftp.FTPFileSystem)
+
+## Class Diagram
+![FTPInputOperator class diagram](images/ftpInputOperator/classdiagram.png)
+
+### Ports
+Because this is an input operator, there are no input ports.
+
+- output: output port on which data is emitted. A downstream operator receives tuples from the FTPInputOpertor
+
+## Partitioning
+#### Static Partitioning
+Configure parameter `partitionCount` to define the desired number of initial partitions
+(4 in this example).
+
+```xml
+<property>
+  <name>dt.operator.{OperatorName}.prop.partitionCount</name>
+  <value>4</value>
+</property>
+```
+#### Dynamic Partitioning
+Dynamic partitioning -- changing the number of partitions of one or more operators
+in a running application -- can be achieved in multiple ways:
+- Use the command line tool `apex` or the UI console to change the value of the
+  `partitionCount` property of the running operator. This change is detected in
+  `processStats()` (which is invoked periodically by the platform) where, if the
+  current partition count (`currentPartitions`) and the desired partition count
+  (`partitionCount`) differ, the `repartitionRequired` flag in the response is set.
+  This causes the platform to invoke `definePartitions()` to create a new set of
+  partitions with the desired count.
+- Override `processStats()` and within it, based on the statistics in the
+  incoming parameter or any other factors, define a new desired value of
+  `partitionCount` and finally, if this value differs from the current partition
+  count, set the `repartitionRequired` flag in the response.
+
+### Configuration
+1. **host**: the hostname of the FTP Server
+2. **source**: the directory path from where to scan and read files.
+3. **username**: the username for authenticating against the FTP server. This is
+an optional property and can be skipped when anonymous FTP is enabled
+4. **password**: the password to be used in conjunction with the above username.
```

Comment:
Show examples of setting these properties both in code as well as in properties files (but please test your examples with an actual running application before adding them).